### PR TITLE
Adds examples for conversion enhancements and form data uploads

### PR DESCRIPTION
**Pull request by jradcliff**
Also adds an `--orderId` parameter to UploadOfflineConversion
since order ID is required for enhancements.

### CODE REVIEW

**Comment by jradcliff** [comment]
This PR moves the example changes made in tg/1173607 (reviewed by Adam and Nick) into `main`.

**Review by aohren — APPROVED** [review]

File: google-ads-examples/src/main/java/com/google/ads/googleads/examples/remarketing/UploadConversionEnhancement.java [hunk 1]
```diff
@@ -0,0 +1,282 @@
...
+
+    // Adds user identifiers, hashing where required.
+
+    // Creates a user identifier using sample values for the user address.
+    UserIdentifier addressIdentifier =
+        UserIdentifier.newBuilder()
+            // User identifiers for conversion enhancements MUST use first party data.
+            .setUserIdentifierSource(UserIdentifierSource.FIRST_PARTY)
+            .setAddressInfo(
+                OfflineUserAddressInfo.newBuilder()
```
Comment (devchas) [inline google-ads-examples/src/main/java/com/google/ads/googleads/examples/remarketing/UploadConversionEnhancement.java:166]: It doesn't look like the OfflineUserAddressInfo builder is ever actually built. Is this correct? Does it work?

File: google-ads-examples/src/main/java/com/google/ads/googleads/examples/remarketing/UploadConversionEnhancement.java [hunk 2]
```diff
@@ -0,0 +1,282 @@
...
+    if (userAgent != null) {
+      // Sets the user agent. This should match the user agent of the request that sent the original
+      // conversion so the conversion and its enhancement are either both attributed as same-device
+      // or both attributed as cross-device.
+      enhancementBuilder.setUserAgent(userAgent);
+    }
+
+    if (restatementValue != null) {
+      // Creates a builder to construct the restated conversion value.
+      RestatementValue.Builder valueBuilder = enhancementBuilder.getRestatementValueBuilder();
```
Comment (devchas) [inline google-ads-examples/src/main/java/com/google/ads/googleads/examples/remarketing/UploadConversionEnhancement.java:199]: It doesn't look like valueBuilder builder is ever actually built. Is this correct? Does it work?

File: google-ads-examples/src/main/java/com/google/ads/googleads/examples/remarketing/UploadConversionEnhancement.java [hunk 3]
```diff
@@ -0,0 +1,282 @@
...
+
+    // Creates the conversion upload service client.
+    try (ConversionAdjustmentUploadServiceClient conversionUploadServiceClient =
+        googleAdsClient.getLatestVersion().createConversionAdjustmentUploadServiceClient()) {
+      // Uploads the enhancement adjustment. Partial failure should always be set to true.
+      UploadConversionAdjustmentsResponse response =
+          conversionUploadServiceClient.uploadConversionAdjustments(
+              UploadConversionAdjustmentsRequest.newBuilder()
+                  .setCustomerId(Long.toString(customerId))
+                  .addConversionAdjustments(enhancementBuilder)
```
Comment (devchas) [inline google-ads-examples/src/main/java/com/google/ads/googleads/examples/remarketing/UploadConversionEnhancement.java:218]: It doesn't look like enhancementBuilder builder was ever actually built. Is this correct? Does it work?

File: google-ads-examples/src/main/java/com/google/ads/googleads/examples/remarketing/UploadConversionEnhancement.java [hunk 4]
```diff
@@ -0,0 +1,282 @@
...
+
+  /**
+   * Runs the example.
+   *
+   * @param googleAdsClient the Google Ads API client.
+   * @param customerId the client customer ID.
+   * @param conversionActionId conversion action ID associated with this conversion.
+   * @param orderId unique order ID (transaction ID) of the conversion.
+   * @param userAgent the HTTP user agent of the conversion.
+   * @param restatementValue the enhancement value
```
Comment (devchas) [inline google-ads-examples/src/main/java/com/google/ads/googleads/examples/remarketing/UploadConversionEnhancement.java:132]: Nit: missing period on this and the following lines.

File: google-ads-examples/src/main/java/com/google/ads/googleads/examples/remarketing/UploadConversionWithIdentifiers.java [hunk 5]
```diff
@@ -0,0 +1,256 @@
...
+    @Parameter(names = ArgumentNames.CUSTOMER_ID, required = true)
+    private long customerId;
+
+    @Parameter(names = ArgumentNames.CONVERSION_ACTION_ID, required = true)
+    private long conversionActionId;
+
+    @Parameter(names = ArgumentNames.EMAIL_ADDRESS, required = true)
+    private String emailAddress;
+
+    @Parameter(names = ArgumentNames.ORDER_ID, required = false)
```
Comment (devchas) [inline google-ads-examples/src/main/java/com/google/ads/googleads/examples/remarketing/UploadConversionWithIdentifiers.java:50]: Nit: I'd change the order to make this last to be consistent with the params order below.

File: google-ads-examples/src/main/java/com/google/ads/googleads/examples/remarketing/UploadConversionWithIdentifiers.java [hunk 6]
```diff
@@ -0,0 +1,256 @@
...
+    }
+
+    try {
+      new UploadConversionWithIdentifiers()
+          .runExample(
+              googleAdsClient,
+              params.customerId,
+              params.conversionActionId,
+              params.emailAddress,
+              params.orderId,
```
Comment (devchas) [inline google-ads-examples/src/main/java/com/google/ads/googleads/examples/remarketing/UploadConversionWithIdentifiers.java:101]: Same nit regarding ordering of params to make orderId last.

File: google-ads-examples/src/main/java/com/google/ads/googleads/examples/remarketing/UploadConversionWithIdentifiers.java [hunk 7]
```diff
@@ -0,0 +1,256 @@
...
+  }
+
+  /**
+   * Runs the example.
+   *
+   * @param googleAdsClient the Google Ads API client.
+   * @param customerId the client customer ID.
+   * @param conversionActionId conversion action ID associated with this conversion.
+   * @param emailAddress email address for the conversion.
+   * @param orderId the unique ID (transaction ID) of the conversion.
```
Comment (devchas) [inline google-ads-examples/src/main/java/com/google/ads/googleads/examples/remarketing/UploadConversionWithIdentifiers.java:127]: Same nit regarding ordering.

File: google-ads-examples/src/main/java/com/google/ads/googleads/examples/remarketing/UploadConversionWithIdentifiers.java [hunk 8]
```diff
@@ -0,0 +1,256 @@
...
+   * @param conversionDateTime date and time of the conversion.
+   * @param conversionValue the value of the conversion.
+   */
+  // [START upload_conversion_with_identifiers]
+  private void runExample(
+      GoogleAdsClient googleAdsClient,
+      long customerId,
+      long conversionActionId,
+      String emailAddress,
+      String orderId,
```
Comment (devchas) [inline google-ads-examples/src/main/java/com/google/ads/googleads/examples/remarketing/UploadConversionWithIdentifiers.java:137]: Same nit regarding ordering.

**Review by devchas — APPROVED** [review]

**Pull request merged**